<p>
The use of the <code>continue</code> branching statement increase the essential complexity of the source code
and so prevent any refactoring of this source code to replace all well structured control structures with a single statement.
</p>

<h2>Noncompliant Code Example</h2>

<pre>
mylabel: for (i = 0 ; i &lt; 3; i++) {
  for (j = 0; j &lt; 4 ; j++) {
    doSomething();
    if (checkSomething()) {
      continue mylabel; // Non-Compliant
    }
  }
}
</pre>

<h2>See</h2>
<ul>
<li>MISRA C:2004, 14.5</li>
</ul>
